pkg/cmd/docs-issue-generation/docs_issue_generation.go

```diff
@@ -15,112 +15,263 @@ import (
+	releaseJustificationRE = regexp.MustCompile(`(?i)release justification:.*`)
+	prNumberRE             = regexp.MustCompile(`Related PR: \[?https://github.com/cockroachdb/cockroach/pull/(\d+)\D`)
+	commitShaRE            = regexp.MustCompile(`Commit: \[?https://github.com/cockroachdb/cockroach/commit/(\w+)\W`)
+)
+
+const (
+	docsOrganization = "cockroachdb"
+	docsRepo         = "docs"
+)
+
 // the heart of the script to fetch and manipulate all data and create the individual docs issues
 func docsIssueGeneration(params parameters) {
-	var search ghSearch // a search for all PRs based on a given commit SHA
-	var prs []pr
-	err := httpGet(
-		"https://api.github.com/search/issues?q=sha:"+params.Sha+"+repo:cockroachdb/cockroach+is:merged",
-		params.Token,
-		&search,
-	)
+	repoID, labelMap, err := searchDocsRepoLabels(params.Token)
 	if err != nil {
-		log.Fatal(err)
-	}
-	prs = prNums(search)     // populate slice of PRs of type pr
-	for _, pr := range prs { // for each PR in the list, get the merge branch and the list of eligible commits
-		var pull ghPull
-		err := httpGet(
-			"https://api.github.com/repos/cockroachdb/cockroach/pulls/"+strconv.Itoa(pr.number),
-			params.Token,
-			&pull,
-		)
-		if err != nil {
-			log.Fatal(err)
+		fmt.Println(err)
+	}
+	prs, err := searchCockroachPRs(params.StartTime, params.EndTime, params.Token)
+	if err != nil {
+		fmt.Println(err)
+	}
+	docsIssues := constructDocsIssues(prs)
+	if params.DryRun {
+		fmt.Printf("Start time: %#v\n", params.StartTime.Format(time.RFC3339))
+		fmt.Printf("End time: %#v\n", params.EndTime.Format(time.RFC3339))
+		if len(docsIssues) > 0 {
+			fmt.Printf("Dry run is enabled. The following %d docs issue(s) would be created:\n", len(docsIssues))
+			fmt.Println(docsIssues)
+		} else {
+			fmt.Println("No docs issues need to be created.")
 		}
-		pr.mergeBranch = pull.Base.Ref
-		var commits []ghPullCommit
-		err = httpGet(
-			"https://api.github.com/repos/cockroachdb/cockroach/pulls/"+strconv.Itoa(pr.number)+"/commits?per_page:250",
-			params.Token,
-			&commits,
-		)
+	} else {
+		for _, di := range docsIssues {
+			if params.DryRun {
+				fmt.Printf("%#v\n", di)
+			} else {
+				err := di.createDocsIssues(params.Token, repoID, labelMap)
+				if err != nil {
+					fmt.Println(err)
+				}
+			}
+		}
+	}
+}
+
+func searchDocsRepoLabels(token string) (string, map[string]string, error) {
+	var labels = map[string]string{}
+	var repoID string
+	repoID, hasNextPage, nextCursor, err := searchDocsRepoLabelsSingle("", labels, token)
+	if err != nil {
+		fmt.Println(err)
+		return "", nil, err
+	}
+	for hasNextPage {
+		_, hasNextPage, nextCursor, err = searchDocsRepoLabelsSingle(nextCursor, labels, token)
 		if err != nil {
-			log.Fatal(err)
+			fmt.Println(err)
+			return "", nil, err
+		}
+	}
+	return repoID, labels, nil
+}
+
+func searchDocsRepoLabelsSingle(
+	cursor string, m map[string]string, token string,
+) (string, bool, string, error) {
```

Comment:
Same comment as above: it is unclear what the return values of this function mean. Note you can [name the return values](https://go.dev/tour/basics/7), but another alternative is just adding a comment above the function.

```
func searchDocsRepoLabelsSingle(
	cursor string, m map[string]string, token string,
) (repoID string, hasNextPage bool, nextCursor string, err error) {
```

Reply:
Done!